include/silo/storage/column/indexed_string_column.h

```diff
@@ -74,19 +80,29 @@ class IndexedStringColumn {
    template <class Archive>
    [[maybe_unused]] void serialize(Archive& archive, const uint32_t /* version */) {
       // clang-format off
+      archive & column_name;
       archive & lookup;
       archive & lineage_tree;
       // clang-format on
    }
 
+   std::string column_name;
    common::BidirectionalMap<std::string> lookup;
    std::optional<common::LineageTree> lineage_tree;
    std::deque<IndexedStringColumnPartition> partitions;
 
   public:
-   IndexedStringColumn();
+   explicit IndexedStringColumn(std::string column_name);
+
+   IndexedStringColumn(
+      std::string column_name,
+      const common::LineageTreeAndIdMap& lineage_tree_and_id_map
+   );
 
-   IndexedStringColumn(const common::LineageTreeAndIdMap& lineage_tree_and_id_map);
+   IndexedStringColumn(const IndexedStringColumn& other) = delete;
+   IndexedStringColumn(IndexedStringColumn&& other) = delete;
+   IndexedStringColumn& operator=(const IndexedStringColumn& other) = delete;
+   IndexedStringColumn& operator=(IndexedStringColumn&& other) = delete;
```

Comment:
This is the column where the contained partition have pointers to some shared data structure. This will make this much more resilient to future changes, which might stumble that. Implicit moves of the data structure could easily lead to unwanted bugs.